src/group/managed_group/mod.rs

```diff
@@ -687,20 +709,20 @@ impl<'a> ManagedGroup<'a> {
     /// MLSCiphertext first.
     fn plaintext_to_mls_messages(
         &mut self,
-        plaintext_messages: Vec<MLSPlaintext>,
-    ) -> Vec<MLSMessage> {
-        plaintext_messages
-            .into_iter()
-            .map(
-                |plaintext| match self.configuration().handshake_message_format {
-                    HandshakeMessageFormat::Plaintext => MLSMessage::Plaintext(plaintext),
-                    HandshakeMessageFormat::Ciphertext => {
-                        let ciphertext = self.group.encrypt(plaintext);
-                        MLSMessage::Ciphertext(ciphertext)
-                    }
-                },
-            )
-            .collect()
+        mut plaintext_messages: Vec<MLSPlaintext>,
```

Comment:
Just to confirm: You just re-factored this so that you can throw an error?

Reply:
yes 🙂 The other option is to collect a `Vec<Result<...>>`, which I'm not a huge fan off.